spec/directives/import/nested.hrx

```diff
@@ -0,0 +1,125 @@
+<===> scope/function/input.scss
+.parent {
+  // This should be visible to the imported stylesheet. There's not really a
+  // good reason for this, but it's the historical behavior so whatever.
+  @function local() {
+    @return value;
+  }
+
+  @import 'other';
+}
+
+<===> scope/function/other.scss
+x {
+  function: local();
+}
+
+<===> scope/function/output.css
+.parent x {
+  function: value;
+}
+
+<===>
+================================================================================
+<===> scope/mixin/input.scss
+.parent {
+  // This should be visible to the imported stylesheet. There's not really a
+  // good reason for this, but it's the historical behavior so whatever.
+  @mixin local {
+    x {y: z}
+  }
+
+  @import 'other';
+}
+
+<===> scope/mixin/other.scss
+@include local;
+
+<===> scope/mixin/output.css
+.parent x {
+  y: z;
+}
+
+<===>
+================================================================================
+<===> scope/variable/input.scss
+.parent {
+  // This should be visible to the imported stylesheet. There's not really a
+  // good reason for this, but it's the historical behavior so whatever.
+  $var: value;
+  @import 'other';
+}
+
+<===> scope/variable/other.scss
+x {
+  var: $var;
+}
+
+<===> scope/variable/output.css
+.parent x {
+  var: value;
+}
+
+<===>
+================================================================================
+<===> at_rule/keyframes/input.scss
+a {@import "other"}
+
+<===> at_rule/keyframes/_other.scss
+// This should ignore the parent selector, since Sass knows @keyframes is only
+// valid at the root of a document.
+@keyframes b {
+  0% {c: d}
+}
+
+<===> at_rule/keyframes/output.css
+@keyframes b {
+  0% {
+    c: d;
+  }
+}
+
+<===>
+================================================================================
+<===> at_rule/childless/input.scss
+a {@import "other"}
+
+<===> at_rule/childless/_other.scss
+@b c;
+
+<===> at_rule/childless/output.css
+a {
+  @b c;
+}
+
+<===>
+================================================================================
+<===> at_rule/declaration_child/input.scss
+a {@import "other"}
+
+<===> at_rule/declaration_child/_other.scss
+@b {c: d}
+
+<===> at_rule/declaration_child/output.css
+@b {
```

Comment:
I'm not familiar with the expectation here but just want to confirm that the `@b` should be moved to top level here but not when it is childless?

Reply:
That's right. An at-rule with children is assumed to behave similar to `@media`/`@supports` *et al* where browsers expect it to appear at the root and contain style rules.